OutSystems/outsystems-maps
# RGRIDT-1009: Create the logic to close popups when other markers get clicked

This PR is for RGRIDT-1009: Create the logic to close popups when other markers get clicked

### What was happening
* There was in fact a need to close all the popups when other markers were getting clicked -> [details](https://www.outsystems.com/forums/discussion/72892/close-current-marker-popup-when-clicking-new-marker/)
* The first solution was to create an API that would close all the popups. This approach was not the best to maintain. 
* Therefore, we opted to change the logic of the infoWindow. 

### What was done
* The solution passes by creating a new feature → InfoWindow. 
* This feature contains all the logic to open/close/set the content of the popup, but instead of being associated with each marker popup (one instance per marker block), this solution has only one infoWindow instantiated on the OSMap and its events are triggered by the marker blocks.

### Test Steps
Test cases | Test steps
-- | --
Open Marker Popup by Click | 1) Click on the Marker in Boston US<br>**Expected: The Marker in Boston US should have a popup.**
Open Marker Popup by API | 1) Click on the Button "Open Popup Boston"<br>**Expected: The Marker in Boston US should have a popup.**
Close Marker Popup by Click | 1) Click on the Marker in Boston US.<br>2) Click on the "x" inside the Popup of the Marker in Boston US.<br>**Expected: The Marker in Boston US should not have a popup.**
Close Marker Popup by API | 1) Click on the Button "Open Popup Boston"<br>2) Click on the Button "Close Popup Boston"<br>**Expected: The Marker in Boston US should not have a popup.**
ESC Marker Popup | 1) Click on the Marker in Boston US.<br>2) Press "ESC" key.<br>**Expected: The Marker in Boston US should not have a popup.**
Refresh Marker Popup | 1) Click on the Marker in Boston US<br>2) The popup should have an image and the text "OutSystems Boston55 Thomson Pl, Boston, MA 02210, United States value = 0"<br>3) Edit the TextArea with "<div style="background-color:red">Test</div>"<br>4) Click on the button "Refresh Popup Boston"<br>**Expected: The Marker in Boston US should have a popup with "Test" and background of the text should be red.**
Check If Marker Popup works with OS expressions | 1) The var should have the value 0. (check the value above the button "Increment")<br>2) Click on the Marker in Boston US<br>3) The popup should have "value = 0" on it.<br>4) Click on the button "Increment Var".<br>5) Close the Popup on the "x".<br>6) Open the Popup again.<br>**Expected: The Marker in Boston US should have a popup with "value = 1" on it.**
Click Event + Open Marker Popup | 1) Click on the Marker in Boston US<br>**Expected: The sample has a feedback message "You clicked on the marker with id: MarkerBoston" and the Marker in Boston US should have a popup**
Platform “if” false | 1) Click on the Marker in Boston US<br>2) Toggle OFF the "Show Marker Boston".<br>**Expected: The Map should not show any marker on Boston US.**
Platform “if” true | 1) Click on the Marker in Boston US<br>2) Toggle OFF the "Show Marker Boston".<br>3) Toggle ON the "Show Marker Boston".<br>**Expected: The Map should show the marker on Boston US. Popup is closed.**
Setting the position | 1) Click on the Marker in Boston US<br>2) Change the Marker Position to "Madrid, Spain" in the input.<br>**Expected: The Map should show the marker in Madrid, Spain and the Popup should still appear on it.**
Change marker icon | 1) Click on the Marker in Boston US<br>2) Change the IconURL in the dropdown to "Parking".<br>**Expected: The Map should show the new marker Icon in Boston US. Popup should still appear on it.**
Open multiple marker popups | 1. Click on "Boston" Marker and wait for the popup.<br>2. Click on "Paris" Marker and wait for the popup.<br>**Expected: The popup hover "Paris" should open and the "Boston" popup should close.**


### Checklist
